### PR TITLE
Jobsearch filter

### PR DESCRIPTION
**Pull request by DennisVanEe**
Added the job search filter. All the links looked to work on my end. Rebased with recent master branch.

### CODE REVIEW

File: homeyess/templates/job_board/job_board.html [hunk 1]
```diff
@@ -11,36 +11,44 @@
...
-						<td>{{ job.location }}</td>
-						<td>{{ job.wage }}</td>
-						<td>{{ job.short_summary }}</td>
-			    	</tr>
-				   {% endfor %}
-			</table>
-		{% endif %}
+	<form method="get" style="text-align: center">
+    	{{ form }}
+    	<button type="submit">Apply Filters</button>
```
Comment (aHunsader) [inline homeyess/templates/job_board/job_board.html]: please use bootstrap button classes: `btn`, `btn-primary`

File: homeyess/website/forms.py [hunk 2]
```diff
@@ -135,6 +135,9 @@ class Meta:
     interview_address = BootstrapCharField(max_length=200, required=True)
     interview_company = BootstrapCharField(max_length=100, required=True)
 
+class JobBoardFilterForm(Form):
+    location = forms.CharField(required=False)
+    job_title = forms.CharField(required=False)
```
Comment (aHunsader) [inline homeyess/website/forms.py]: please use class `BootstrapCharField` so that we can style all fields at once

File: homeyess/website/views.py [hunk 3]
```diff
@@ -176,8 +175,28 @@ def job_board(request):
...
+    jobs = filterQuerySetJobs(
+        job_posts,
+        location,
+        job_title)
+
+    context = {}
+    context['form'] = JobBoardFilterForm(initial={
+        'location': request.GET.get('location', None),
+        'job_title': request.GET.get('job_title', None),
+    })
```
Comment (aHunsader) [inline homeyess/website/views.py]: i like this

File: homeyess/website/views.py [hunk 4]
```diff
@@ -392,6 +411,28 @@ def confirm_ride(request, ride_id):
...
+    # If no filter has been applied, don't filter:
+    if location is None and job_title is None:
+        return jobs
+
+    # This way we can filter if only one of the values is set:
+    loc_filter = location.lower() if location is not None else ''
+    title_filter = job_title.lower() if job_title is not None else ''
+
+    return list(filter(lambda job: (loc_filter in job.location.lower()) and (title_filter in job.job_title.lower()), jobs))
+
```
Comment (aHunsader) [inline homeyess/website/views.py]: please put this in a file `job_utils.py` in the same directory. I am doing the same for ride util functions

File: homeyess/templates/job_board/job_board.html [hunk 5]
```diff
@@ -11,36 +11,44 @@
...
+  	</form>
+	<br>
+	<div class="flex-container" style="margin-left: 100px; margin-right: 100px; min-height: 400px;">
```
Comment (aHunsader) [inline homeyess/templates/job_board/job_board.html]: why are we using flex-container stuff? Can we use bootstrap instead to keep it uniform?

**Comment by DennisVanEe** [comment]
I made most of the changes requested. For the last comment regarding the flex-container stuff, how should I use bootstrap here? (Not really sure what bootstrap is...). I just copied the html from ride_board.html.

**Comment by aHunsader** [comment]
lgtm

**Pull request merged**